The new icu package does not build

The new icu update does not build. The fix is updating config.{guess,sub} as suggested. Here is the log:

latte [posh] C:\src\vcpkg > .\vcpkg.exe install icu --triplet x64-windows
The following packages will be built and installed:
    icu:x64-windows
Building package icu:x64-windows...
-- CURRENT_INSTALLED_DIR=C:/src/vcpkg/installed/x64-windows
-- DOWNLOADS=C:/src/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/src/vcpkg/packages/icu_x64-windows
-- CURRENT_BUILDTREES_DIR=C:/src/vcpkg/buildtrees/icu
-- CURRENT_PORT_DIR=C:/src/vcpkg/ports/icu/.
-- Using cached C:/src/vcpkg/downloads/icu4c-59_1-src.zip
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Applying patch C:/src/vcpkg/ports/icu/disable-escapestr-tool.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch C:/src/vcpkg/ports/icu/disable-escapestr-tool.patch done
-- Configuring x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message):
    Command failed: C:/src/vcpkg/downloads/tools/msys2/msys64/usr/bin/bash.exe;--noprofile;--norc;-c;C:/src/vcpkg/buildtrees/i
cu/src/icu-59.1/icu/source/runConfigureICU MSYS/MSVC --host=i686-pc-mingw32 --disable-samples --disable-tests --disable-static
 --enable-shared --disable-debug --enable-release --prefix=C:/src/vcpkg/packages/icu_x64-windows
    Working Directory: C:/src/vcpkg/buildtrees/icu/x64-windows-rel
    See logs for more information:
      C:\src\vcpkg\buildtrees\icu\configure-x64-windows-rel-out.log
      C:\src\vcpkg\buildtrees\icu\configure-x64-windows-rel-err.log

Call Stack (most recent call first):
  ports/icu/portfile.cmake:55 (vcpkg_execute_required_process)
  scripts/ports.cmake:73 (include)


Error: Building package icu:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: icu:x64-windows
  Vcpkg version: 0.0.80-89ad7d6b22e71ed058c2386c147154603da4bee9

Additionally, attach any relevant sections from the log files above.
latte [posh] C:\src\vcpkg > cat "C:\src\vcpkg\buildtrees\icu\configure-x64-windows-rel-err.log"
C:/src/vcpkg/buildtrees/icu/src/icu-59.1/icu/source/config.guess: unable to guess system type

This script, last modified 2013-11-29, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (C:/src/vcpkg/buildtrees/icu/src/icu-59.1/icu/source/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2013-11-29

uname -m = x86_64
uname -r = 2.6.0(0.304/5/3)
uname -s = MSYS_NT-10.0
uname -v = 2016-09-07 20:45

/usr/bin/uname -p = unknown
/bin/uname -X     =

hostinfo               =
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              = x86_64
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = x86_64
UNAME_RELEASE = 2.6.0(0.304/5/3)
UNAME_SYSTEM  = MSYS_NT-10.0
UNAME_VERSION = 2016-09-07 20:45
config.sub: too many arguments
Try `config.sub --help' for more information.
configure: error: /bin/sh C:/src/vcpkg/buildtrees/icu/src/icu-59.1/icu/source/config.sub dummy.c
Microsoft (R) Incremental Linker Version 14.10.25019.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:dummy.exe
/out:C:/src/vcpkg/downloads/tools/msys2/msys64/tmp/cgzZDbMR/dummy.exe
dummy.obj  failed
latte [posh] C:\src\vcpkg > cat "C:\src\vcpkg\buildtrees\icu\configure-x64-windows-rel-out.log"
export CPP= CC=cl CXX=cl CPPFLAGS= CFLAGS=-Gy -MD -MD -O2 -Oi -Zi CXXFLAGS=-Gy -MD -MD -O2 -Oi -Zi LDFLAGS=-DEBUG -INCREMENTAL
:NO -OPT:REF -OPT:ICF MAKE=
Running ./configure  --host=i686-pc-mingw32 --disable-samples --disable-tests --disable-static --enable-shared --disable-debug
 --enable-release --prefix=C:/src/vcpkg/packages/icu_x64-windows for MSYS using Microsoft Visual C++ compiler

checking for ICU version numbers... release 59.1, library 59.1, unicode version 9.0
checking build system type... C:/src/vcpkg/buildtrees/icu/src/icu-59.1/icu/source/runConfigureICU: ./configure failed

--- a/ports/icu/CONTROL
+++ b/ports/icu/CONTROL
@@ -1,3 +1,3 @@
 Source: icu
-Version: 59.1
+Version: 59.1-1
 Description: Mature and widely used Unicode and localization library.

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -24,9 +24,12 @@ set(ENV{PATH} "${NEWPATH}")
 set(BASH ${MSYS_ROOT}/usr/bin/bash.exe)
 
 vcpkg_execute_required_process(
-    COMMAND ${BASH} --noprofile --norc -c "pacman -Sy --noconfirm --needed make"
+    COMMAND ${BASH} --noprofile --norc -c "pacman -Sy --noconfirm --needed make automake1.15"
     WORKING_DIRECTORY "${MSYS_ROOT}"
     LOGNAME "pacman-${TARGET_TRIPLET}")
+
+set(AUTOMAKE_DIR ${MSYS_ROOT}/usr/share/automake-1.15)
+file(COPY ${AUTOMAKE_DIR}/config.guess ${AUTOMAKE_DIR}/config.sub DESTINATION ${SOURCE_PATH}/source)
 
 set(CONFIGURE_OPTIONS "--host=i686-pc-mingw32 --disable-samples --disable-tests")
 

